monitoring/opentelemetry-reactive/src/test/java/io/quarkus/ts/opentelemetry/reactive/MetricsIT.java

```diff
@@ -0,0 +1,110 @@
+            .withProperty("quarkus.application.name", "pongservice");
+
+    @QuarkusApplication(classes = { PingResource.class, PingPongService.class, AdminResource.class }, dependencies = {
+            // Required for metrics. TODO: replace when OTLP gets its own metrics provider
+            @Dependency(artifactId = "quarkus-micrometer-registry-prometheus")
+    })
+    static final RestService pingservice = new RestService()
+            .withProperty("pongservice.url", () -> pongservice.getURI(HTTP).getRestAssuredStyleUri())
+            .withProperty("pongservice.port", () -> Integer.toString(pongservice.getURI(HTTP).getPort()))
+            // verify OTEL service name has priority over default Quarkus application name
+            .withProperty("quarkus.otel.service.name", OTEL_PING_SERVICE_NAME);
+
+    @BeforeAll
+    static void beforeAll() {
+        pingPongRequest();
+        pingPongRequest();
+    }
+
+    private static void pingPongRequest() {
+        given().when()
+                .get(pingservice.getURI(HTTP).withPath("/ping/pong").toString())
+                .then()
+                .statusCode(HttpStatus.SC_OK).body(equalToIgnoringCase("ping pong"));
+    }
+
+    @Test
+    @Tag("QUARKUS-5424")
+    public void testExemplars() {
+        await().ignoreExceptions().atMost(30, TimeUnit.SECONDS)
+                .pollInterval(5, TimeUnit.SECONDS)
+                .untilAsserted(() -> {
+                    Response response = pingservice.given().get("/q/metrics");
+                    assertEquals(HttpStatus.SC_OK, response.statusCode());
+                    final String metricName = "http_server_requests_seconds_count";
+                    final String body = response.body().asString();
+                    String metric = Arrays.stream(body.split("\n"))
+                            .filter(line -> line.startsWith(metricName))
+                            .findFirst()
+                            .orElseThrow(() -> new AssertionError(metricName + " was not found in " + body));
+                    String[] content = metric.split(" ");
+                    assertEquals(6, content.length, "Some values are missing from " + metric);
+                    assertEquals("""
+                            http_server_requests_seconds_count{method="GET",outcome="SUCCESS",status="200",uri="/ping/pong"}
+                            """.strip(), content[0]);
+
+                    assertEquals("2.0", content[1], "Amount of events is wrong");
+                    assertEquals("#", content[2], "Unexpected exemplar separator!");
+                    assertTrue(content[3].contains("span_id"), "Exemplar doesn't contain span ID");
+                    assertTrue(content[3].contains("trace_id"), "Exemplar doesn't contain trace ID");
+                    assertEquals("1.0", content[4], "Unexpected exemplar value!");
+                });
+    }
+
+    @Test
+    @Tag("QUARKUS-5637")
+    public void testNetty() {
```

Comment:
AFAICT this test verifies that metrics doesn't have high cardinality label as id because has too many values, I'd probably call it differently, something like `testNettyIdLabelNotPresent` or whatever suits you